### PR TITLE
375 sorting (II) 

### LINKED ISSUE #375: 🚀 [Feature] Sorting
> [ulfgebhardt](https://github.com/ulfgebhardt) **Authored by [ulfgebhardt](https://github.com/ulfgebhardt)**
_<time datetime="2019-04-03T11:44:54Z" title="Wednesday, April 3rd 2019, 1:44:54 pm +02:00">Apr 3, 2019</time>_
_Closed <time datetime="2019-07-31T12:49:38Z" title="Wednesday, July 31st 2019, 2:49:38 pm +02:00">Jul 31, 2019</time>_
---

### Sorting

image

- [ ] Latest (newest)
- [ ] Oldest
- [ ] Most Popular (shouts)
- [ ] (optional) Most commented/interactions https://github.com/human-connection/human-connection/issues/265

### Userstory

missing @roschaefer 

### Design & Layout - Alpha

Closed:
image

Open:
image

### Design & Layout - Nitro
missing

### PR DESCRIPTION
**Pull request by Tirokk**
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-07-17T07:58:50Z" title="Wednesday, July 17th 2019, 9:58:50 am +02:00">Jul 17, 2019</time>_
_Merged <time datetime="2019-07-31T12:49:38Z" title="Wednesday, July 31st 2019, 2:49:38 pm +02:00">Jul 31, 2019</time>_
---

## 🍰 Pullrequest
contributions can be sorted 
- createdAt_desc > (Newest)
- createdAt_asc > (Oldest)
- shoutedCount_desc > (Popular)
- commentsCount_desc > (Commented)

the sorting also works with the filter bubble

ezgif com-video-to-gif (3)
### Issues
- fixed #375
  
### Todo
 - [X] Frontend & logic 
-  [X] backend

### CODE REVIEW

**Comment by Tirokk** [comment]
> [ogerly](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-07-19T11:09:33Z" title="Friday, July 19th 2019, 1:09:33 pm +02:00">Jul 19, 2019</time>_

---


@mattwr18  or @Tirokk I would be grateful if you would correct the mistakes. 
i get unfortunately on my system the errors not yet reproduced

**Comment by Tirokk** [comment]
> [appinteractive](https://github.com/appinteractive) **Authored by [appinteractive](https://github.com/appinteractive)**
_<time datetime="2019-07-22T13:14:41Z" title="Monday, July 22nd 2019, 3:14:41 pm +02:00">Jul 22, 2019</time>_

---


Hey from the UX perspective it makes no sense to have a searchable select there. We should use a dropdown here as it will likely never exceed a couple of entries.

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-24T08:23:20Z" title="Wednesday, July 24th 2019, 10:23:20 am +02:00">Jul 24, 2019</time>_

---


> Hey from the UX perspective it makes no sense to have a searchable select there. We should use a dropdown here as it will likely never exceed a couple of entries.

I agree with you @appinteractive, but there is not a dropdown from the styleguide which is not searchable, correct? Are you suggesting using just HTML/CSS?

**Comment by Tirokk** [comment]
> [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-24T09:39:29Z" title="Wednesday, July 24th 2019, 11:39:29 am +02:00">Jul 24, 2019</time>_

---


Oh right, we have the `Dropdown.vue` component, :stuck_out_tongue:

File: webapp/locales/de.json [hunk 1]
```diff
@@ -23,6 +23,12 @@
     "bank": "Bankverbindung",
     "germany": "Deutschland"
   },
+  "sorting": {
+    "newest": "Neuste",
+    "oldest": "Älteste",
+    "popular": "Beliebt",
+    "commented": "meist Kommentiert"
```
Comment (Tirokk) [inline webapp/locales/de.json:30]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-31T12:32:56Z" title="Wednesday, July 31st 2019, 2:32:56 pm +02:00">Jul 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,6 +23,12 @@
     "bank": "Bankverbindung",
     "germany": "Deutschland"
   },
+  "sorting": {
+    "newest": "Neuste",
+    "oldest": "Älteste",
+    "popular": "Beliebt",
+    "commented": "meist Kommentiert"
```


I think -> „Meist Kommentiert"

File: webapp/locales/en.json [hunk 2]
```diff
@@ -23,6 +23,12 @@
     "bank": "bank account",
     "germany": "Germany"
   },
+  "sorting": {
+    "newest": "Newest",
+    "oldest": "Oldest",
+    "popular": "Popular",
+    "commented": "most Commented"
```
Comment (Tirokk) [inline webapp/locales/en.json:30]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-31T12:33:29Z" title="Wednesday, July 31st 2019, 2:33:29 pm +02:00">Jul 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -23,6 +23,12 @@
     "bank": "bank account",
     "germany": "Germany"
   },
+  "sorting": {
+    "newest": "Newest",
+    "oldest": "Oldest",
+    "popular": "Popular",
+    "commented": "most Commented"
```


I think -> „Most Commented"

File: webapp/pages/index.vue [hunk 3]
```diff
@@ -9,6 +9,17 @@
           @clearSearch="clearSearch"
         />
       </ds-flex-item>
+      <ds-flex-item>
+        <div class="sorting-dropdown">
+          <ds-select
+            v-model="selected"
+            :options="sortingOptions"
+            size="large"
```
Comment (Tirokk) [inline webapp/pages/index.vue:17]: > [Tirokk](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-31T12:35:09Z" title="Wednesday, July 31st 2019, 2:35:09 pm +02:00">Jul 31, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -9,6 +9,17 @@
           @clearSearch="clearSearch"
         />
       </ds-flex-item>
+      <ds-flex-item>
+        <div class="sorting-dropdown">
+          <ds-select
+            v-model="selected"
+            :options="sortingOptions"
+            size="large"
```


The typo is a little bit to big, in my eyes.

**Review by Mogge — APPROVED** [review]
Wowoww @ogerly !!! 🚀💫

Stunning work my dear. Thank you so much.
We love you 😍

Cat and Dog cuddling

**Pull request closed without merging**